tests/foreman/performance/test_pulp_concurrent_initial_sync.py

```diff
@@ -0,0 +1,103 @@
+        cls.savepoint = "20150602-savepoint1-password"
+
+        cls.num_iterations = 0 # depend on # of threads
+        cls.bucket_size = 0 # depend on # of iterations each thread will evenly conduct
+
+        # parameters for getting enabled repositories information
+        cls.repo_ids_list = []
+        cls.repo_names_list = []
+        cls.org_id = 1
+
+        # add date to log and csv files
+        today = date.today()
+        cls.raw_file_name = "{0}-{1}".format(today.strftime('%Y%m%d'), cls.raw_file_name) 
+        cls.stat_file_name = "{0}-{1}".format(today.strftime('%Y%m%d'), cls.stat_file_name)
+
+    def setUp(self):
+        super(ConcurrentTestCase, self).setUp()
+
+        """Restore database before concurrent deleting subscribed systems"""
+        self._restore_from_savepoint(self.savepoint)
+
+        """Retrieve repository list information"""
+        (self.repo_ids_list, self.repo_names_list) = self._get_enabled_repos()
+
+
+    '''Get all enabled repositories ids and names'''
+    def _get_enabled_repos(self):
+        self.logger.info('---------------------------------')
+        self.logger.info('Get All enabled repositories from database:')
+
+        result = Repository.list({'organization-id':cls.org_id}, per_page=False)
+        # repo_ids_list would contain all ids in the hammer repo list
+        # print result.stdout
+        repo_ids_list = [repo['id'] for repo in result.stdout]
+        repo_names_list = [repo['name'] for repo in result.stdout]
+        return (repo_ids_list, repo_names_list)
+
+    def test_sync_2_threads(self):
+       """
+        @Test: Delete subscriptions concurrently using 2 threads
+
+        @Steps:
+        1. get list of all enabled repositories (setup)
+        2. create result dictionary
+        3. create a list of thread names
+        4. concurrent run by multiple threads and each thread delete only a sublist of repos
+        5. produce result of timing
+
+        @Assert: Restoring from database would have 5k registered systems initially
+        """        
+        current_num_threads = self.num_threads[2]
+
+
+        """Parameter for test 2 threads"""
+        
+
+
+        # Create a list to store all threads
+        thread_list = []
+        # Create a dictionary to store all timing results from each thread
+        time_result_dict = {}
+
+        '''
+        Create new threads and start the thread which has a sublist of repos
+        '''
+        for i in range(current_num_threads):
+            time_result_dict["thread-{}".format(i)] = []
+            thread = InitialSyncPerfThread(i, "thread-{}".format(i),
+                                       repo_ids_list[:i], time_result_dict)
+
+        thread.start()
+        thread_list.append(thread)
+
+
+        """wait all threads in thread list"""
+        self._join_all_threads(thread_list)
+
+        """write raw result of sync"""
+
+
+        """write stat result of sync"""
```

Comment:
If some method is incomplete and broken, do not include it in Robottelo at all. Push the code in to some other branch and make a GitHub issue requesting that the feature be completed.